src/black/strings.py

```diff
@@ -0,0 +1,319 @@
+"""
+String-related utility functions.
+"""
+
+import regex as re
+import sys
+import tempfile
+from typing import List, Pattern
+
+from mypy_extensions import mypyc_attr
+
+
+STRING_PREFIX_CHARS = "furbFURB"  # All possible string prefix characters.
+
+
+def sub_twice(regex: Pattern[str], replacement: str, original: str) -> str:
+    """Replace `regex` with `replacement` twice on `original`.
+
+    This is used by string normalization to perform replaces on
+    overlapping matches.
+    """
+    return regex.sub(replacement, regex.sub(replacement, original))
+
+
+def format_hex(text: str) -> str:
```

Comment:
This is for numeric literals, not strings. It should go wherever `normalize_numeric_literals` went.